src/api/app/jobs/send_event_emails.rb

```diff
@@ -7,7 +7,12 @@ class SendEventEmails
   def perform
     Event::Base.where(mails_sent: false).order(created_at: :asc).limit(1000).each do |event|
       subscribers = event.subscribers
-      next if subscribers.empty?
+
+      if subscribers.empty?
+        event.update_attributes(mails_sent: true)
+        next
+      end
```

Comment:
Yes unfortunately the logic for determine the subscribers for an event is very complex and determined by this class and the logic only works one way: we can find subscribers for an event, but not events for a subscriber (or no subscriber).

https://github.com/evanrolfe/open-build-service/blob/fix%2Fproject_log_rotate/src/api/app/models/event_find_subscriptions.rb#L1